x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypes.java

```diff
@@ -62,32 +60,6 @@ public static DataType fromName(String name) {
         return type != null ? type : UNSUPPORTED;
     }
 
-    public static DataType fromJava(Object value) {
-        if (value == null) {
-            return NULL;
-        }
-        if (value instanceof Boolean) {
-            return BOOLEAN;
-        }
-        if (value instanceof Integer) {
-            return INTEGER;
-        }
-        if (value instanceof Long) {
-            return LONG;
-        }
-        if (value instanceof Double) {
-            return DOUBLE;
-        }
-        if (value instanceof Float) {
-            return FLOAT;
-        }
-        if (value instanceof String || value instanceof Character || value instanceof BytesRef) {
-            return KEYWORD;
-        }
-
-        return null;
-    }
-
     public static boolean isUnsupported(DataType type) {
         return DataType.isUnsupported(type);
     }
```

Comment:
Should this be removed? Not sure if it brings any value now.

Reply:
Also, `isString()` and `isDateTimeOrTemporal()` are dups of the same functionality now in `DataType` and `isPrimitive()` has confusingly different implementation than there. 
I guess you're considering just dropping this class at some point, but noting, just in case.

Reply:
Yeah, squashing them together at least.